airgun/entities/host_new.py

```diff
@@ -32,6 +38,163 @@ def get_details(self, entity_name, widget_names=None):
         self.browser.plugin.ensure_page_safe()
         return view.read(widget_names=widget_names)
 
+    def edit_system_purpose(
+        self, entity_name, role=None, sla=None, usage=None, release_ver=None, add_ons=None
+    ):
+        """
+        Function that edits the system purpose of a host
+
+        Args:
+            entity_name: Name of the host
+            role: Role to be assigned
+            sla: SLA to be assigned
+            usage: Usage to be assigned
+            release_ver: Release version to be assigned
+            add_ons: Add-ons to be assigned
+
+        Raises:
+            ValueError: If no parameters are passed.
+        """
+
+        if not any([role, sla, usage, release_ver, add_ons]):
+            raise ValueError(
+                'At least one of the role, sla, usage, release_ver, add_ons ' 'must be provided!'
+            )
+        view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
+        view.wait_displayed()
+        view.overview.system_purpose.edit_system_purpose.click()
+        view = EditSystemPurposeView(self.browser)
+        self.browser.plugin.ensure_page_safe()
+        view.wait_displayed()
+        if role:
+            view.role.fill(role)
+        if sla:
+            view.sla.fill(sla)
+        if usage:
+            view.usage.fill(usage)
+        if release_ver:
+            view.release_version.fill(release_ver)
+        if add_ons:
```

Comment:
Do we need to be so explicit with the conditions here? Can't we just call `fill` on the whole form with `fill` ignoring everything that's not defined? Isn't that what we do elsewhere?

Reply:
@lhellebr After investigating for a while I have found out that this form is not implemented as a PF form (I couldn't even find basic PF Form implementation in widgetastic though) but as an individual PF FormSelects which needs to be filled one by one. I have also tried to look for some implementation of these forms by someone else, but everything I have found (i.e.  `views/host.py -> class general()`) is done in the same way as mine is.

Reply:
You're the expert in new hosts interface so I'll leave it up to you but look at this and consider something similar: https://github.com/SatelliteQE/airgun/blob/master/airgun/views/subscription.py#L108 and invocation https://github.com/SatelliteQE/airgun/blob/master/airgun/entities/subscription.py#L129